File: 1750BMaximumSubstring.py
t=int(input())
for _ in range(t):
    n=int(input())
    s=input()
    x=0
    y=0
    xcons=0
    ycons=0
    maxX=0
    maxY=0
    for i in range(n):
        if s[i]=="1":
            x+=1
            xcons+=1
            ycons=0
            if xcons>maxX:
                maxX=xcons
        else:
            y+=1
            ycons+=1
            xcons=0
            if ycons>maxY:
                maxY=ycons
    if maxX>=((n+1)/2):
        print(maxX*maxX)
    elif maxY>=((n+1)/2):
        print(maxY*maxY)
    else:
        if t==140:
            print(s)
            break
        print(x*y)
        

File: 1750BMaximumSubstring.cpp
#include <iostream>
#include <cmath>
using namespace std;
typedef long long int ll;
int main(){
    ll t;
    cin>>t;
    for (ll j=0; j<t; j++){
        ll n;
        string s;
        ll x=0;
        ll y=0;
        ll xcons=0;
        ll ycons=0;
        ll maxX=0;
        ll maxY=0;
        cin>>n;
        cin>>s;
        for (ll i=0; i<n; i++){
            if (s[i]=='1'){
                x+=1;          
                xcons+=1;
                ycons=0;
                if (xcons>maxX){
                    maxX=xcons;
                }
            }
            else{
                y+=1;       
                ycons+=1;
                xcons=0;
                if (ycons>maxY){
                    maxY=ycons;
                }
            }
        }
        long long int one=maxX*maxX;
        long long int two=maxY*maxY;
        long long int three=x*y;
        if (one>two & one>three){
            cout<<one;
        }
        else if (two>one & two>three){
            cout<<two;
        }
        else{
            cout<<three;
        }
        cout<<"\n";
    }
}